src/Services/DataService/Epics/FetchRecordEpic.test.ts

```diff
@@ -0,0 +1,761 @@
+      expect(spies.complete.callCount).to.equal(1, "it should complete immediately");
+    });
+
+    it("should return separate buffered observables for contexts of different names", () => {
+      const firstId = random.number().toString();
+      const secondId = random.number().toString();      
+      const timeoutPeriod = DEFAULT_COALESCE_BUFFER_TIME + 2;
+
+      const firstPerformRequest = performBufferedRequest(context);
+      const newContext = {
+        ...context,
+        name: random.word(),
+      };
+      const secondPerformRequest = performBufferedRequest(newContext);
+      
+      firstPerformRequest(firstId);
+      const sut = secondPerformRequest(secondId);
+      let action;
+      sut
+        .take(1)
+        .subscribe(a => action = a);
+
+      return new Promise((resolve, reject) => {
+        setTimeout(() => {
+          try {
+            expect(action).to.deep.equal({
+              type: "PUSH_RECORD",
+              payload: {
+                fullText: expectedResult.fullText,
+                id: secondId,
+              },
+            });
+            resolve();
+          } catch (e) {
+            reject(e);
+          }
+        }, timeoutPeriod);
+      });
+    });
+
+    it("should subscribe to a new buffered observable for contexts of the same name when requested after the timeout period", () => {
+      const id = random.number().toString();
+      const secondId = random.number().toString();
+      const firstTimeOutPeriod = DEFAULT_COALESCE_BUFFER_TIME + 1;
+      const secondTimeOutPeriod = firstTimeOutPeriod * 2;
+
+      const performRequest = performBufferedRequest(context);
+
+      performRequest(id)
+        .take(1)
+        .subscribe();
+
+      let action;
+      setTimeout(() => {
+        performRequest(secondId)
+          .take(1)
+          .subscribe(a => action = a);
+      }, firstTimeOutPeriod);
+
+      return new Promise((resolve, reject) => {
+        setTimeout(() => {
+          try {
+            expect(action).to.deep.equal({
+              type: "PUSH_RECORD",
+              payload: {
+                fullText: expectedResult.fullText,
+                id: secondId,
+              },
+            });
+            resolve();
+          } catch (e) {
+            reject(e);
+          }
+        }, secondTimeOutPeriod);
+      });
+    });
+
+    it("should complete the buffered observable after it emits once to prevent memory leaks", () => {
+      const id = random.number().toString();
+
+      const spies = {
+        next: spy(),
+        complete: spy(),
+      };
+
+      const performRequest = performBufferedRequest(context);      
+
+      performRequest(id)
+        .subscribe(spies);
+
+      return new Promise((resolve, reject) => {
+        setTimeout(() => {
+          try {
+            expect(spies.complete.callCount).to.equal(1, "it should complete immediately");
```

Comment:
writing "it should complete immediately" here seems wrong since we are expecting it to complete after the timeout